src/main/java/jenkins/security/plugins/ldap/LDAPConfiguration.java

```diff
@@ -577,6 +615,10 @@ public ApplicationContext createApplicationContext(LDAPSecurityRealm realm) {
         vars.put("com.sun.jndi.ldap.read.timeout", "60000"); // timeout if no response after 60 seconds
         vars.putAll(getExtraEnvVars());
         contextSource.setBaseEnvironmentProperties(vars);
+
+        // pooled connections do not work with StartTLS
+        contextSource.setPooled(false);
```

Comment:
interesting. so you mean `com.sun.jndi.ldap.connect.pool` doesn't work anymore? it creates connectivity issues or just ignored? If so there are few usage this property in the code.

Reply:
yeah, this one took me a while to figure out, but every example i was able to find on the internet related to StartTLS and Spring had `setPooled(false)`

> Explicitly disable connection pooling, or Spring may attempt to StartTLS twice on the same connection.

Also there seems to be some known issues as documented in the `spring-ldap` docs: https://docs.spring.io/spring-ldap/docs/1.3.2.RELEASE/reference/html/pooling.html